app/assets/javascripts/components/story/ExpandedStory/ExpandedStoryControls.js

```diff
@@ -1,33 +1,52 @@
-import React from 'react';
-
-const ExpandedStoryControls = ({ onCancel, onSave, onDelete, readOnly }) => {
-  return (
-    <div className="form-group Story__controls">
-      <input className="save"
-        onClick={onSave}
-        type="button"
-        value={I18n.t('save')}
-        disabled={readOnly}
-      />
-
-      <input className="delete"
-        onClick={() => {
-          if (window.confirm(I18n.t('story destroy confirm'))){
-            onDelete();
-          }
-        }}
-        type="button"
-        value={I18n.t('delete')}
-        disabled={readOnly}
-      />
-
-      <input className="cancel"
-        onClick={onCancel}
-        type="button"
-        value={I18n.t('cancel')}
-      />
-    </div>
-  );
+import React, { Component } from 'react';
+
+class ExpandedStoryControls extends Component {
+  constructor(props) {
+    super(props);
+
+    this.handleCancel = this.handleCancel.bind(this);
+  }
+
+  handleCancel() {
+    const { onCancel, isDirty } = this.props;
+
+    if (isDirty && !window.confirm(I18n.t('story unsaved changes'))) {
```

Comment:
Don't you prefer to abstract the `!window.confirm(I18n.t('story unsaved changes'))` in a function or variable? It's a bit harder to figure out what it is doing.